spring-integration-zip/src/main/java/org/springframework/integration/zip/splitter/UnZipResultSplitter.java

```diff
@@ -32,21 +33,37 @@
  * @since 1.0
  *
  */
-public class UnZipResultSplitter {
-
-	public List<Message<Object>> splitUnzippedMap(Map<String, Object> unzippedEntries) {
+public final class UnZipResultSplitter {
 
-		final List<Message<Object>> messages = new ArrayList<Message<Object>>(unzippedEntries.size());
+	public List<Message<Object>> splitUnzippedMap(Message<Map<String, Object>> message) {
+		return createMessages(message.getPayload(), message.getHeaders());
+	}
 
+	private List<Message<Object>> createMessages(Map<String, Object> unzippedEntries, MessageHeaders headers) {
+		List<Message<Object>> messages = createList(unzippedEntries.size());
 		for (Map.Entry<String, Object> entry : unzippedEntries.entrySet()) {
-			final String path = FilenameUtils.getPath(entry.getKey());
-			final String filename = FilenameUtils.getName(entry.getKey());
-			final Message<Object> splitMessage = MessageBuilder.withPayload(entry.getValue())
-					.setHeader(FileHeaders.FILENAME, filename)
-					.setHeader(ZipHeaders.ZIP_ENTRY_PATH, path).build();
-			messages.add(splitMessage);
+			messages.add(createMessage(entry, headers));
 		}
 		return messages;
 	}
 
+	private static List<Message<Object>> createList(int initialCapacity) {
+		return new ArrayList<Message<Object>>(initialCapacity);
+	}
+
+	private static Message<Object> createMessage(Map.Entry<String, Object> entry, Map<String, ?> headers) {
+		return MessageBuilder.withPayload(entry.getValue())
+                .setHeader(FileHeaders.FILENAME, getName(entry))
+                .setHeader(ZipHeaders.ZIP_ENTRY_PATH, getPath(entry))
+				.copyHeaders(headers)
+                .build();
+	}
+
+	private static String getName(Map.Entry<String, Object> entry) {
+		return FilenameUtils.getName(entry.getKey());
+	}
+
+	private static String getPath(Map.Entry<String, Object> entry) {
+		return FilenameUtils.getPath(entry.getKey());
```

Comment:
@artembilan the purpose of these methods isn't general. It's not utility methods like e.g. `Math#max`. These methods are small atom pieces of algorithm (get file name and get file path from zipped entry). Moreover, again, with adding `protected` and removing `static` you can provide this methods for class successor for modification in the future. E.g. get file name may extract the name in non standard format (not in format provided by `FilenameUtils` by default implementation) or even get file name from entry value.

Again if you think more about [Premature Optimization](http://c2.com/cgi/wiki?PrematureOptimization) than flexibility and maintainability then... well... remove them. And I'll make note about this as the project style in my future code changes :)